.github/workflows/go.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: Go
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions: {}
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
```

Comment:
hmm, can we pin dependencies? I think dependabot does that by default but would be a good practice. (e.g. https://github.com/guacsec/guac/blob/main/.github/workflows/ci.yaml#L43)

Reply:
Done.